### PR TITLE
add hw reastaurnt

### PR DESCRIPTION
**Pull request by dymekkqa**

### CODE REVIEW

File: HW_12/src/bar.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        this.orders.push(obj);
+    }
+
+    removeOrderById(id: number): void {
+        for (let i = 0; i < this.orders.length; i++) {
+            if (this.orders[i].orderId === id) {              
+              this.orders.splice(i, 1); 
+              break; 
+            }
+          }
```
Comment (anatoly-karpovich) [inline HW_12/src/bar.ts]: maybe findIndex?) and handle if id is not found with new Error

File: HW_12/src/order.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { Meal } from "./meel";
+
+export class Order {
+    customerName: string;
+    orderItems: Meal[];
+    private totalPrice: number;
+    orderId?: number;
```
Comment (anatoly-karpovich) [inline HW_12/src/order.ts]: What for orderId here if you have id in IBarOrder?

**Review by anatoly-karpovich — APPROVED** [review]
Well Done!

File: HW_12/src/beer.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { Meal } from "./meel";
+import { beerValue, beerBrand} from "./enums"
+
+export interface IBeer {
+    beerValue: beerValue;
+    beerBrand: beerBrand;
+    calculatePrice(): number;
```
Comment (VladimirLQA) [inline HW_12/src/beer.ts]: This method is already declared in the abstract class `Meal`

File: HW_12/src/order.ts (hunk 2 above)
Comment (dymekkqa) [reply HW_12/src/order.ts]: it was a missclick, i used fast fixing by IDE. Fixed the issue

File: HW_12/src/beer.ts (hunk 3 above)
Comment (dymekkqa) [reply HW_12/src/beer.ts]: u a right

**Review by anatoly-karpovich — APPROVED** [review]

**Pull request merged**